src/charm.py

```diff
@@ -396,6 +427,219 @@ def _on_client_changed(self, event: ClientChangedEvent) -> None:
             event.defer()
             return
 
+    def _on_create_oauth_client_action(self, event: ActionEvent) -> None:
+        if not self._hydra_service_is_running:
+            event.fail("Service is not ready.")
```

Comment:
Can we consider a more indicative message, for example `Service is not ready. Please re-run the action when the charm is active`?